packages/core/src/lib/implementation/execute-plugin.ts

```diff
@@ -105,18 +114,34 @@ export async function executePlugin(
       ...(description && { description }),
       ...(docsUrl && { docsUrl }),
       ...(groups && { groups }),
-    };
-    return pluginReport;
+    } satisfies PluginReport;
   } catch (error) {
     const e = error as Error;
     throw new PluginOutputError(slug, e);
   }
 }
 
+const MOCK_PROGRESS: ProgressBar = {
+  incrementInSteps: (_: number) => {
+    _;
+    void 0;
+  },
+  updateTitle: (_: string) => {
+    _;
+    void 0;
+  },
+  endProgress: (_?: string) => {
+    _;
+    void 0;
+  },
+};
```

Comment:
This confused me, it seems like it shouldn't be in production code, because it's named `MOCK_` and has a very "stubby" implementation. Maybe the intent would be a bit more clear like this?

```suggestion
const NOOP_PROGRESS: ProgressBar = {
  incrementInSteps: () => { /* do nothing */ },
  updateTitle: () => { /* do nothing */ },
  endProgress:() => { /* do nothing */ },
};
```